### PR TITLE
Add feature to hold pins during deep sleep.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tests/esp32/README.md [hunk 1]
```diff
@@ -22,16 +22,20 @@ On board 1 connect as follows:
...
 3. IO23 - IO22
 
+IO16 must stay unconnected.
+
 ## Running
 
 As of 2023-05-30.
 Run the tests individually. All test run on board 1, except for the
-uart, ultrasound, and wait_for tests. (See their respective files for more info.)
+uart, ultrasound, wait_for tests, and pin-hold tests. (See their respective files for more info.)
```
Comment (kasperl) [inline tests/esp32/README.md:38]:
```suggestion
uart, ultrasound, wait-for tests, and pin-hold tests. (See their respective files for more info.)
```

File: src/compiler/propagation/type_primitive_esp32.cc [hunk 2]
```diff
@@ -40,6 +40,10 @@ TYPE_PRIMITIVE_ANY(memory_page_report)
 TYPE_PRIMITIVE_ANY(watchdog_init)
 TYPE_PRIMITIVE_ANY(watchdog_reset)
 TYPE_PRIMITIVE_ANY(watchdog_deinit)
+TYPE_PRIMITIVE_ANY(pin_hold_enable)
```
Comment (kasperl) [inline src/compiler/propagation/type_primitive_esp32.cc:43]: These really only return null.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/compiler/propagation/type_primitive_esp32.cc (hunk 2 above)
Comment (floitsch) [reply src/compiler/propagation/type_primitive_esp32.cc:43]: Agreed.
There wasn't any 'NULL' variant in the file and I didn't look for one. I figured that nobody would use the result anyway.
Changed the watchdog and pin-hold functions to be typed as null.
done.

**Pull request merged**